### PR TITLE
Add version check at connection

### PR DESCRIPTION
**Pull request by VLitanADI**
implements issue #466

### CODE REVIEW

File: apps/server/server.cpp [hunk 1]
```diff
@@ -620,6 +621,12 @@ void invoke_sdk_api(payload::ClientRequest buff_recv) {
         break;
     }
```
Comment (dNechita) [inline apps/server/server.cpp:623]: Why do you need the server to tell what connection type it is? It can’t be anything else than Network.

**Comment by VLitanADI** [comment]
TODO after discussion with @dNechita :

- [ ] use hacky include instead of exposing `connection_validator` to the clients of the sdk
- [ ] rename `getConnectionString` to something more suggestive: `getConnectionToken`, `getVersionString` ?

File: apps/server/server.cpp (hunk 1 above)
Comment (dNechita) [reply apps/server/server.cpp:623]: Ignore this. I misunderstood things.

File: sdk/src/connections/network/network_depth_sensor.cpp [hunk 2]
```diff
@@ -85,6 +86,7 @@ NetworkDepthSensor::~NetworkDepthSensor() {
 aditof::Status NetworkDepthSensor::open() {
     using namespace aditof;
 
+    std::string connectionString;
```
Comment (dNechita) [inline sdk/src/connections/network/network_depth_sensor.cpp:89]: This doesn't seem to be used anywhere, nor the connection_validator.h.

Comment (VLitanADI) [reply sdk/src/connections/network/network_depth_sensor.cpp]: removed it.

File: sdk/src/connections/network/network_sensor_enumerator.cpp [hunk 3]
```diff
@@ -44,18 +45,57 @@ NetworkSensorEnumerator::NetworkSensorEnumerator(const std::string &ip)
...
 
     if (net->ServerConnect(m_ip) != 0) {
         LOG(WARNING) << "Server Connect Failed";
         return Status::UNREACHABLE;
     }
 
+    getVersionString(net, connectionString);
+
+    if (!isValidConnection(aditof::ConnectionType::NETWORK, connectionString)) {
+        LOG(ERROR) << "invalid connection string: " << connectionString;
```
Comment (dNechita) [inline sdk/src/connections/network/network_sensor_enumerator.cpp:95]: now, it's version string. Update err message and variable name

**Review by dNechita — APPROVED** [review]

**Comment by dNechita** [comment]
You need to fix build on Windows before merging!

**Pull request merged**